### ISSUE TITLE
[Bug]: Milvus2.4 crash after I search on empty bf16 collection

### ISSUE THREAD

**Issue by shanghaikid** [milvus-io/milvus#32301]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.4-20240413-e00772a0-amd64
```

### Current Behavior

Milvus2.4 crash after I search on empty bf16 collection.

![img_v3_029v_501ebe74-5ef4-470a-81fc-ee1fe7499a5g](https://github.com/milvus-io/milvus/assets/185051/a7a4e24a-7275-4e2e-9f4f-203bfaf04662)

### Steps To Reproduce

```markdown
1. create a collection with bf16
2. create index
3. load index
4. search without insert any data
```

### Milvus Log

https://grafana-4am.zilliz.cc/explore?orgId=1&panes=%7B%22evb%22:%7B%22datasource%22:%22vhI6Vw67k%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Bpod%3D%5C%22ruiyimastertest24-hhfuj-milvus-standalone-6b89dfbc67-cj8sj%5C%22%7D%22,%22hide%22:true,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22vhI6Vw67k%22%7D,%22editorMode%22:%22code%22,%22queryType%22:%22range%22%7D,%7B%22refId%22:%22B%22,%22expr%22:%22%7Bpod%3D%5C%22ruiyimastertest24-hhfuj-milvus-standalone-6b89dfbc67-cj8sj%5C%22%7D%7C~%5C%22%5C%22%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22vhI6Vw67k%22%7D,%22editorMode%22:%22code%22,%22queryType%22:%22range%22%7D%5D,%22range%22:%7B%22from%22:%22now-3h%22,%22to%22:%22now%22%7D%7D%7D&schemaVersion=1

**Comment by congqixia** [comment]
shall be fixed by #32220 #32258 please verify with latest image in branch 2.4
/assign @shanghaikid 
/unassign @cqy123456

**Comment by shanghaikid** [comment]
verified on the latest image.
